lib/Service/PageService.php

```diff
@@ -971,8 +971,10 @@ public function matchBacklinks(PageInfo $pageInfo, string $content): bool {
 		$suffix = '\)/';
 
 		$protocol = 'https?:\/\/';
-		$trustedDomainConfig = (array)$this->config->getSystemValue('trusted_domains', []);
-		$trustedDomains = $trustedDomainConfig !== [] ? '(' . implode('|', $trustedDomainConfig) . ')' : 'localhost';
+		$trustedDomainArray = array_map(static function (string $domain) {
+			return preg_quote($domain, '/');
+		}, (array)$this->config->getSystemValue('trusted_domains', []));
```

Comment:
`preg_quote` will quote `*` and thus remove the special meaning it has inside `trusted_domains`. So for example with a trusted domains of `*.my.tld` the resulting regexp would fail to find a absolute link using `www.my.tld` as the domain name.

I'm pretty sure this did not work before either - but still, while we're at it...

I wonder if replacing `\*` with `\w*` after the preg replace would do the trick.

Reply:
Edit: ah, just saw that you did this :blush:

Reply:
> I'm not sure if wildcards in trusted domains would even be supported

docs say so:

![grafik](https://github.com/nextcloud/collectives/assets/97337118/34c667ae-6c14-4a72-92cd-84de263cd9e6)

Reply:
https://docs.nextcloud.com/server/27/admin_manual/configuration_server/config_sample_php_parameters.html?highlight=trusted_domains